apps/website/src/pages/index.astro

```diff
@@ -435,4 +445,28 @@ const currentPage = new URL(Astro.request.url).pathname;
       }
     }
   </style>
+
+  <script>
+    const updateVisibility = () => {
+      const visibleBottomLine = window.scrollY + window.innerHeight;
+      const demoWrappers = document.querySelectorAll('.move-on-visible');
+      demoWrappers.forEach((demoWrapper) => {
+        if (!(demoWrapper instanceof HTMLElement)) {
+          return;
+        }
+
+        const top = demoWrapper.getBoundingClientRect().top;
+        if (top < visibleBottomLine) {
+          demoWrapper.dataset.visible = 'true';
+        } else {
+          delete demoWrapper.dataset.visible;
+        }
+      });
+    };
+
+    updateVisibility();
+    document.querySelector('body').addEventListener('scroll', () => {
```

Comment:
have you tried using intersection observer instead of scroll events? it's a lot more efficient

Reply:
I initially did that, but then it was even unbluring and moving when scrolling from bottom to top. I felt that was too distracting. So with the current approach where it only animates on scrolling from top to bottom.

But now that I think of it, I think if I use an intersection observer itself and check the scroll position in the callback, it will give the same behavior too. I will try that.

Reply:
Used `IntersectionObserver` instead. https://github.com/iTwin/iTwinUI/pull/1032/files/0f4e8e178270263fd4366a316a5acb3c66bb582a..c464aa1dc29ddda3ec3606bf6786cacafb5998ab

Reply:
can you increase the [threshold](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API#thresholds) a little bit? right now, even if 1% of the section is visible, it will animate.

for example, just the top of the "why itwinui" is visible in this viewport size, so it was animating when the page loads, before even scrolling
![](https://user-images.githubusercontent.com/9084735/215119317-f3c84a4a-659d-4b26-a263-dd23cec30b8e.png)

Reply:
Added a threshold of `[0.3]`